aws/src/test/java/org/apache/iceberg/aws/RESTSigV4SignerTest.java

```diff
@@ -0,0 +1,114 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.aws;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockserver.model.Header.header;
+import static org.mockserver.model.HttpRequest.request;
+
+import java.io.IOException;
+import java.util.Map;
+import org.apache.hc.core5.http.HttpHeaders;
+import org.apache.hc.core5.http.HttpStatus;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.rest.HTTPClient;
+import org.apache.iceberg.rest.auth.OAuth2Util;
+import org.apache.iceberg.rest.responses.ConfigResponse;
+import org.apache.iceberg.rest.responses.OAuthTokenResponse;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockserver.integration.ClientAndServer;
+import org.mockserver.model.HttpResponse;
+import software.amazon.awssdk.auth.signer.internal.SignerConstant;
+
+public class RESTSigV4SignerTest {
+  private static ClientAndServer mockServer;
+  private static HTTPClient client;
+
+  @BeforeClass
+  public static void beforeClass() {
+    mockServer = ClientAndServer.startClientAndServer();
+
+    System.setProperty("aws.accessKeyId", "fake_id");
+    System.setProperty("aws.secretAccessKey", "fake_secret");
+
+    Map<String, String> properties =
+        ImmutableMap.of(
+            CatalogProperties.URI,
+            "http://localhost:" + mockServer.getLocalPort(),
+            "http.request.interceptor-impl",
+            RESTSigV4Signer.class.getName());
+    client = HTTPClient.buildFrom(properties);
+  }
+
+  @AfterClass
+  public static void afterClass() throws IOException {
+    mockServer.stop();
+    client.close();
+  }
+
+  @Test
+  public void signRequestWithoutBody() {
+    mockServer
+        .when(
+            request()
+                .withMethod("GET")
+                .withPath("/v1/config")
+                .withHeader(header(HttpHeaders.AUTHORIZATION, "AWS4-HMAC-SHA256.*")))
+        .respond(HttpResponse.response().withStatusCode(HttpStatus.SC_OK).withBody("{}"));
+
+    ConfigResponse response =
+        client.get("v1/config", ConfigResponse.class, ImmutableMap.of(), e -> {});
+    assertThat(response.defaults()).isEmpty();
```

Comment:
Similar to above, we just want to have something to check that the request was actually posed and responded so that we don't fall through in the event that there's no matching request in the mock server.

Reply:
I updated the assertions to be more inline with expectations